clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/cache/AzureServerGroupCachingAgent.groovy

```diff
@@ -246,7 +126,7 @@ class AzureServerGroupCachingAgent extends AzureCachingAgent {
     }
 
     try {
-      updatedServerGroup = metricsSupport.readData {
+      newServerGroup = metricsSupport.readData {
```

Comment:
it looks like the code always assumes a new server group. what if the OnDemand is on a delete (see eviction)? If the resource is not there we might get a CloudException or the API call if we are lucky might just return null.